### PR TITLE
Refactor/user repo

### PR DESCRIPTION
**Pull request by okalimbet**
## What is the change? 
Changed name convention, adjusted syntax. New `getDateIndex()` and `getAverageOfValues()` were added as helper functions to follow SRP rules. 
New tests were added to test `getDateIndex()` and `getAverageOfValues()` functions.
userrepo-test file was cleaned and refactored for a better readability.

## What does it fix?
Readbility
SRP

## Is this a fix or a feature? 
Fix/refactoring

## Where should the reviewer start? 
User-repo.js:
4 - end:  syntax fix, new helper were functions added

userrepo-test.js:
42 - new extra tests for newly created helper functions

index.html - script added
195
## How should this be tested?
Running npm test should pass all tests

### CODE REVIEW

File: test/userrepo-test.js [hunk 1]
```diff
@@ -479,31 +456,54 @@ describe('User Repo', function() {
...
+
+      expect(getHydrationDataByUserId).to.eql({
         '3': [1]
-      })
+      });
+    });
+    it('should get average data of the provided data', function() {
+      const values = [3.5, 4, 3.3, 3.6, 3.6, 4, 3.1];
+
+      expect(userRepo.getAverageOfValues(values).toFixed(2)).to.eql('3.59')
```
Comment (mdflynn) [inline test/userrepo-test.js:506]: I like how you were able to write this test.

File: src/User-repo.js [hunk 2]
```diff
@@ -1,46 +1,62 @@
...
+  }
+
   getFirstWeek(id, dataSet) {
     return this.makeSortedUserArray(id, dataSet).slice(0, 7);
-  };
+  }
+
+  getDateIndex(date, id, dataSet) {
+    return this.makeSortedUserArray(id, dataSet)
+      .indexOf(this.makeSortedUserArray(id, dataSet)
```
Comment (cameronRomo) [inline src/User-repo.js:37]: First time with chaining. This is a neat feature for JS, glad it passes the test.

**Review by mdflynn — APPROVED** [review]
It was a good idea to change the methods to es6 arrows. Much cleaner to read.

File: src/User-repo.js [hunk 3]
```diff
@@ -49,30 +65,28 @@ class UserRepo {
       return objectSoFar;
     }, {});
   }
+
+  getAverageOfValues(data) {
```
Comment (cameronRomo) [inline src/User-repo.js:69]: Nice, effort here for SRP.

**Review by cameronRomo — APPROVED** [review]
Nice, this PR fixes a lot of issues and significantly drys up the code for `User-repo.js`

**Pull request merged**